WooCommerce/Classes/ViewRelated/Dashboard/DashboardViewModel.swift

```diff
@@ -15,9 +15,28 @@ final class DashboardViewModel {
 
     @Published var announcementViewModel: AnnouncementCardViewModelProtocol? = nil
 
+    var storeOnboardingViewModel: StoreOnboardingViewModel? = nil {
+        didSet {
+            storeOnboardingViewModel?.noTasksAvailableForDisplay
+                .map({ false })
+                .assign(to: &$showOnboarding)
```

Comment:
Assigning multiple publishers, here and in L62 from `userDefaults`, to one value `showOnboarding` could result in an unexpected state like when two streams emit conflicting values.

I'd recommend using `combineLatest` to combine these two publishers so that the visibility can be determined clearly based on these two values. The function that implements the `combineLatest` can early return when `featureFlagService.isFeatureFlagEnabled(.dashboardOnboarding) == false`.

Because [`combineLatest` waits for both values to become available](https://rxmarbles.com/#combineLatest), it might make sense for `noTasksAvailableForDisplay` to be a `CurrentValueSubject` so that it has an initial value. This way, the combined value is available immediately and we can hide the onboarding card if the user defaults publisher says so regardless of the actual `noTasksAvailableForDisplay` value.

Reply:
> Assigning multiple publishers, here and in L62 from userDefaults, to one value showOnboarding could result in an unexpected state like when two streams emit conflicting values.

I agree. 

I took this approach because `DashboardViewModel.storeOnboardingViewModel` will not be available when `DashboardViewModel` is initialized.

1. I cannot create a Publisher using `combineLatest` in the `DashboardViewModel`'s `init` method because `storeOnboardingViewModel` will be created only later. 
1. Using `combineLatest` inside `storeOnboardingViewModel` `didSet` will mean that `showOnboarding` must initially be set as `true`, and we have to initialize `StoreOnboardingViewModel` even if the `userDefaults` value says that we don't have to show the list. This will create unnecessary network requests. 

I will try to think of a solution to use `combineLatest`. Kindly let me know if I misunderstood your suggestion or if you have any other ideas.

Reply:
I started using an `AnyCancellable`, which we can setup during init and reset again when `storeOnboardingViewModel` gets assigned. https://github.com/woocommerce/woocommerce-ios/pull/9185/commits/e82df0d58b65e815558a8149d426f53a199679a4

I will test this further, polish and add unit tests. Kindly let me know if you have early feedback.

Reply:
Just left a suggestion on moving the logic with UserDefaults to `StoreOnboardingViewModel` 🙂